### PR TITLE
Fix NSUrlCredential tests on reference platform

### LINKED ISSUE #774: NSURLCredential tests fail on OSX
[----------] 3 tests from NSURLCredential
[ RUN      ] NSURLCredential.credentialWithUser
[       OK ] NSURLCredential.credentialWithUser (1 ms)
[ RUN      ] NSURLCredential.credentialWithIdentity
Foundation/NSURLCredentialTests.mm:40: Failure
Value of: credential != __null
  Actual: false
Expected: true
FAILED: credential should be non-null!
[  FAILED  ] NSURLCredential.credentialWithIdentity (0 ms)
[ RUN      ] NSURLCredential.credentialForTrust
Foundation/NSURLCredentialTests.mm:55: Failure
Value of: [credential persistence]
  Actual: 1
Expected: NSURLCredentialPersistenceNone
Which is: 0
FAILED: persistence is not valid
[  FAILED  ] NSURLCredential.credentialForTrust (0 ms)
[----------] 3 tests from NSURLCredential (3 ms total)

### PR DESCRIPTION
**Pull request by aballway**
Trying to create a NSURLCredential without an identity should return nil. The default persistence when creating with trust should be NSURLCredentialPersistenceForSession.

Fixes #774

### CODE REVIEW

**Comment by aballway** [comment]
@msft-Jeyaram is added to the review. #Closed

**Comment by ms-jihua** [comment]
:shipit:

**Comment by DHowett-MSFT** [comment]
:shipit:

File: Frameworks/Foundation/NSURLCredential.mm [hunk 1]
```diff
@@ -71,6 +75,7 @@ - (instancetype)initWithIdentity:(SecIdentityRef)identity
 - (instancetype)initWithTrust:(SecTrustRef)trust {
     if (self = [super init]) {
         _trust = trust;
+        _persistence = NSURLCredentialPersistenceForSession;
     }
```
Comment (msft-Jeyaram) [inline Frameworks/Foundation/NSURLCredential.mm]: if(self = [self _initWithPersistence::NSURLCredentialPersistenceForSession]) {
  _trust = trust
}

given we are calling [super init] there anyways.

File: Frameworks/Foundation/NSURLCredential.mm [hunk 2]
```diff
@@ -58,6 +58,10 @@ + (NSURLCredential*)credentialForTrust:(SecTrustRef)trust {
 - (instancetype)initWithIdentity:(SecIdentityRef)identity
                     certificates:(NSArray*)certArray
                      persistence:(NSURLCredentialPersistence)persistence {
```
Comment (msft-Jeyaram) [inline Frameworks/Foundation/NSURLCredential.mm]: could you add test for invalid cases here. nil, etc.
and verify the behavior

**Comment by msft-Jeyaram** [comment]
:clock1:

File: Frameworks/Foundation/NSURLCredential.mm (hunk 2 above)
Comment (aballway) [inline Frameworks/Foundation/NSURLCredential.mm]: We don't have any of this implemented so I don't know how we could test it beyond passing in nil.  Could be made a todo.

---
In reply to: [73749039](https://github.com/Microsoft/WinObjC/pull/826#discussion_r73749039) [](ancestors = 73749039)

**Comment by rajsesh-msft** [comment]
:shipit:

**Comment by bbowman** [comment]
:shipit:

**Pull request merged**